### PR TITLE
Feature/adjust time function

### PR DESCRIPTION
**Pull request by sliu008**
### Description

Update l2ss-py to remove warnings and better get the time variable

### Overview of work done

- Add function to get the time variable for a granule for better temporal subsetting
- Update l2ss-py to remove warnings using sizes instead of dims
- Add script and update github actions to deploy to harmony for release candidates and release
- Add parallelization to pytest

### Overview of verification done

- Tested locally
- Ran pytests locally
- Tested harmony deploy script but didn't make actual deployment

## PR checklist:

* [ ] Linted
* [ ] Updated unit tests
* [ ] Updated changelog
* [ ] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

File: deployment/harmony_deploy.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    return next(iter(tokens))
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Update the service image tag.")
+    parser.add_argument("--tag", help="The new tag version to update.", required=True)
+    args = parser.parse_args()
+
+    url = f"https://harmony.{'uat.' if ENV == 'uat' else ''}earthdata.nasa.gov/service-image-tag/#podaac-l2-subsetter"
```
Comment (jamesfwood) [inline deployment/harmony_deploy.py:50]: If this doesn't work, trying taking out the "#".  I've always ran it without the # in the url.

**Review by jamesfwood — APPROVED** [review]

File: podaac/subsetter/subset.py [hunk 2]
```diff
@@ -542,12 +543,22 @@ def compute_time_variable_name(dataset: xr.Dataset, lat_var: xr.Variable, total_
...
+           ('units' in var.attrs and time_units_pattern.match(var.attrs['units'])))) and var_name not in total_time_vars:
+            return var_name
+
     # then check if any variables have 'time' in the string if the above loop doesn't return anything
     for var_name in list(dataset.data_vars.keys()):
         var_name_time = var_name.strip(GROUP_DELIM).split(GROUP_DELIM)[-1]
         if len(dataset[var_name].squeeze().dims) == 0:
             continue
         if var_name not in total_time_vars and 'time' in var_name_time.lower() and dataset[var_name].squeeze().dims[0] in lat_var.squeeze().dims:
+            print(var_name)
```
Comment (frankinspace) [inline podaac/subsetter/subset.py:561]: change to a log call or remove

File: podaac/subsetter/dimension_cleanup.py [hunk 3]
```diff
@@ -117,8 +117,8 @@ def recreate_pixcore_dimensions(datasets: list):
     dim_dict = {}
     count = 0
     for dataset in datasets:
-        dim_list_shape = list(dataset.dims.values())
-        current_dims = list(dataset.dims.keys())
+        dim_list_shape = list(dataset.sizes.values())
+        current_dims = list(dataset.sizes.keys())
```
Comment (frankinspace) [inline podaac/subsetter/dimension_cleanup.py:121]: Can you explain the change here? Is `dims` deprecated in favor of `sizes`?

Comment (sliu008) [reply podaac/subsetter/dimension_cleanup.py:121]: Moslty future warnings i think there will be deprecation in the future but for now its not deprecated yet

/worker/podaac/subsetter/subset.py:531: FutureWarning: The return type of `Dataset.dims` will be changed to return a set of dimension names in future, in order to be more consistent with `DataArray.dims`. To access a mapping from dimension names to lengths, please use `Dataset.sizes`.\r\n  time_vars = list(filter(lambda var_name: 'time' in var_name, dataset.dims.keys()))\r\n

**Review by frankinspace — APPROVED** [review]

**Pull request merged**